src/Loop.php

```diff
@@ -430,3 +455,10 @@ private function __construct()
         // intentionally left blank
     }
 }
+
+// Reset the $level in order to be able to use ::run() inside a shutdown handler
+$f = function()
+{
+    self::$level = 0;
+};
+register_shutdown_function($f->bindTo(null, Loop::class));
```

Comment:
I pushed a commit dropping the level requirement… since shutdown function order can't be relied upon (and some apps even start the loop in a shutdown function) it just isn't worth trying to enforce it. Only so much we can do to protect users from themselves…